add options to useAuthState hook "onUserChanged"

dohomi opened the pull request:
Add possibility to pass options for onUserChanged, a function which is called if a user auth state changed.
This might close https://github.com/CSFrequency/react-firebase-hooks/issues/33

mauriceackel commented on auth/useAuthState.ts:
```diff
@@ -7,14 +7,20 @@ export type AuthStateHook = LoadingHook<
+         // onUserLoaded function to process custom claims on any other trigger function
+         await options.onUserChanged(user)
+      }
+      setValue(user);
```
This will never be executed if the passed handler fails. It should be considered to set the value first and then perform the handler.

dohomi replied:
@mauriceackel or should it be wrapped in try .. catch ?

dohomi replied:
Alternatively it could be mentioned that the `onUserChanged` should handle errors within that function.

chrisbianca replied:
This definitely needs some error handling built in, otherwise it will be very difficult to debug any issues. I'd propose a try / catch, then re-using `setError`

chrisbianca commented:
Thanks @dohomi and apologies for the slow review of the PR. I think this is an interesting way of solving the problem and can't decide if this or a `useIdTokenResult` hook might be a better approach?

dohomi commented:
@chrisbianca I added a try catch block inside of the function. I followed the documentation of Firebase thats why I added the option into the result of the `onUserChanged`. I am not sure if `useIdTokenResult` is handling it in the same way.

chrisbianca commented:
v5 is now released: https://github.com/CSFrequency/react-firebase-hooks/releases/tag/v5.0.0
